### ISSUE TITLE
Breakpoint not Hit on Remote Subprocess that Spawned a Child Thread

### ISSUE THREAD

**Issue by nurkenn-keda** [microsoft/debugpy#956]
Before creating a new issue, please check the [FAQ](https://github.com/microsoft/debugpy/wiki/FAQ) to see if your question is answered there.

## Environment data

- debugpy version: 1.6 (run `import debugpy; print(debugpy.__version__)` if uncertain)
- OS and version: Debug on Windows 10, Remote machine on Ubuntu 20.04.4 LTS
- Python version (& distribution if applicable, e.g. Anaconda): Python 3.8.10
- Using VS Code or Visual Studio: VS Code

## Actual behavior

Breakpoint is not hit at all

## Expected  behavior

Breakpoint should work such that user can check variables' values, step-in, etc.

## Steps to reproduce:
1. Followed the steps outlined in https://code.visualstudio.com/docs/python/debugging#_remote-script-debugging-with-ssh to be able to connect local debugger to remote machine
2. Remote machine is running application with Odoo15 framework community edition
3. After connecting try putting a breakpoint on any of the user's code

## Additional Info
1. Odoo15 spawned subprocesses to process HTTP in parallel
2. Said subprocesses spawned additional threads via python's threading library
3. Breakpoint actually works on lines before Odoo calls `t.join()`
4. Breakpoint fails to work in the thread's actual code
5. Breakpoints works flawlessly on Odoo14

**Comment by jeffery9** [comment]
#1206

**Comment by jeffery9** [comment]
#1277
